### Reading and displaying images.

In [1]:
# Import opencv and numpy
import cv2
import numpy as np

In [2]:
# Load an image called moin.jpg in the sub directory called images.
input = cv2.imread('./images/glassbox.jpg')

# Display image using imshow. First parameter to imshow is the title to be shown on the image window. Second is the image
#object created using the imread function.
cv2.imshow('Hello World', input)

# waitKey can be used input information to the opened image window.
cv2.waitKey()

# Close all windows
cv2.destroyAllWindows()

### Have a look at how images are stored in numpy.

In [3]:
print(input.shape)

(816L, 1224L, 3L)


It can be intrepreted as folows. There are 816 rows(height in pixels) and 1224 columns(width in pixels).

In [4]:
print("Height of the image is {} pixels".format(int(input.shape[0])))

Height of the image is 816 pixels


In [5]:
print("Width of the image is {} pixels".format(int(input.shape[1])))

Width of the image is 1224 pixels


### Write back the modified image.

In [6]:
# We shall remove green and blue colors and consider only red.
changedImage = input[:, :, 0]

In [7]:
# display the changed image.
cv2.imshow('Hello World', changedImage)
cv2.waitKey()
cv2.destroyAllWindows()

In [9]:
# Write back the modified image.
cv2.imwrite('./images/modifiedGlassbox.jpg', changedImage)

True